.azure-pipelines/templates/build_validation_pipeline.yml

```diff
@@ -1,5 +1,53 @@
+parameters:
+  - name: is_docker
+    displayName: "Is Docker used for build validation?"
+    type: boolean
+    default: false
+  - name: image_name
+    displayName: "Name of the Docker image"
+    type: string
+  - name: model_type
+    displayName: "Model Type"
+    type: string
+
 jobs:
+- job: Build_Validation_Pipeline_Docker
+  condition: eq(${{ parameters.is_docker }}, true)
+  steps:
+  - script: |
+      python -m pip install --upgrade pip
+      pip install -r .azure-pipelines/requirements/build_validation_requirements.txt
+      flake8 .
+    displayName: 'Install dependencies and run linting'
+
+  - script: | 
+      echo "Building Docker container for validation with tag: ${{ parameters.image_name }}"
+      docker build -t ${{ parameters.image_name }} -f mlops/${{ parameters.model_type }}/environment/Dockerfile mlops/${{ parameters.model_type }}/environment
```

Comment:
Please find time to have a design conversation with @sbaidachni on this approach for enabling container images to be used for CI and CD operations. Worthwhile to think if this is acceptable to go in as is (with any GH issues for optimizations/enhancements), or if we want to invest time in considering alternatives.

Reply:
Resolving for now as this has approvals to be merged in. Will make a new GH issue to discuss this.